### ISSUE TITLE
Initial idea and variants.

### ISSUE THREAD

**Issue by TheHockeyist** [TheHockeyist/99-bottles#1]
I've heard a lot of variants of the song. Some people start with 100 or some start with 99.

The beverage varies depending on the version. I learned it as milk (e.g. I sing "99 bottles of milk on the wall"...) but traditionally, it's an alcoholic beverage that many people would rather not have their children singing about. So I decided to include an option for choosing your beverage of choice.

The third line also varies, from "Take one down, pass it around" to "If one of those bottles should happen to fall".

And lastly, what happens when we reach zero?

0 bottles of milk on the wall,
0 bottles of milk.
Go to the store and buy some more,
99 bottles of milk on the wall!

That's how I learned it.

**Comment by TheHockeyist** [comment]
Does the code work? I think it works, but I'm not printing everything to make sure.

**Comment by benajaero** [comment]
```
Traceback (most recent call last):
  File "python", line 4
    set bottles = 99
              ^
SyntaxError: invalid syntax
```

In both python 2.7 and python 3.

BTW: Nice to see you again

**Comment by TheHockeyist** [comment]
Workarounded it.

EDIT: Okay, I think I probably fixed it.

**Comment by benajaero** [comment]
You need to commit for me to view it. Also it seems like the repeat code is broken. I'm not a python expert though

**Comment by TheHockeyist** [comment]
Ah, okay.

Any other people want to help me out here?

**Comment by benajaero** [comment]
@theinitializer codes in python

**Comment by TheHockeyist** [comment]
Ah, okay. Not sure if he knows that we exist here, but I could use a bit of help.

**Comment by benajaero** [comment]
I mentioned him

**Comment by TheHockeyist** [comment]
Okay.

**Comment by benajaero** [comment]
Question: Why did you leave scratch again?

**Comment by TheHockeyist** [comment]
Project got remixed, disputed it with the creator, creator refused to delete it, took it to the ST.

Then while talking, there was a dispute with the ST. Nothing I could do.

Infuriated, I reassessed why I came to Scratch and if the community was the same as it was when I joined. It became more immature, I concluded Scratch was insufficient for my needs, and I left.

**Comment by benajaero** [comment]
ok

**Comment by TheInitializer** [comment]
Hello, what is this? I got mentioned.

Is there anything you need help with?

**Comment by TheHockeyist** [comment]
Yeah, I need some help with coding this. I'm not sure if my code works or not.

EDIT: Typo.

**Comment by matthewr6** [comment]
1.  `set` has nothing to do with setting variables in Python
2.  `==` is for comparison, and `=` is for setting variables.
3.  `repeat X` is not valid in Python - you want to do `for i in range(X)`
4.  `print` requires commas between things (or you could combine them with the `+` operator)
5.  Learn Python's syntax first https://docs.python.org/3/

**Comment by TheInitializer** [comment]
Haha. 

Well - not sure how else to say this - it's all wrong. None of that is valid Python (aside from the first two lines, lol).

I'd advise you take a quick Python tutorial and learn how it works before writing any serious code 😛

**Comment by towerofnix** [comment]
Since people are raiding this thread and talking about how not-Python the code is.. I recommend you learn to use IDLE so you can test your code as you're working on it, and also [GitHub Desktop](https://desktop.github.com/), so you can actually commit your code and make that code appear online (or git, if you're cool and used to the command line).

You'll find it a *lot* more useful in the long run when you start working on bigger projects.

**Comment by TheHockeyist** [comment]
I attempted to somewhat Python-ize the code.

I'm coming from Scratch, so some of the syntax is still unfamiliar.

**Comment by matthewr6** [comment]
Again, `set` has nothing to do with variable assignment.  You should probably read the syntax docs and frequently refer to them as you write this

**Comment by TheHockeyist** [comment]
Reattempted.

Yeah, I have no idea what I'm doing. :-/.

**Comment by TheInitializer** [comment]
It'll probably be more useful to, you know, actually *learn Python* before making anything with it, as opposed to just trying to guess what to do :P

**Comment by TheHockeyist** [comment]
Well, it's my first time really coding something independently in Python.

We all have to start somewhere.

**Comment by matthewr6** [comment]
And the best place to start is to:
> *learn Python* before making anything with it, as opposed to just trying to guess what to do :P

**Comment by towerofnix** [comment]
Er, learn the *basics* of Python before making anything with it. Just enough to open up IDLE and see some real results on your screen – you know, just learn how to use the `print` statement or function and *see* the result. Then move on from there. The whole "see it on your screen" part is.. really, really important.. :tada:

edit: is it not more fun to see the results on your screen, as you're programming the code that makes those results? :)

**Comment by benajaero** [comment]
Good advice. I think you should try to learn Py 3 not 2, though.
https://docs.python.org/3/tutorial/

@TheInitializer Are you ignoring me?

**Comment by TheInitializer** [comment]
@Blaze349 is correct, there's really not much of a reason to stick with Python 2 anymore. Python 3 is the way to go at this point.

@Blaze349 wdym

**Comment by matthewr6** [comment]
@TheInitializer legacy (or "legacy") code that's a pain to port

**Comment by TheInitializer** [comment]
I doubt OP will be working with much of that though

**Comment by benajaero** [comment]
@TheInitializer nvm

@Firedrake969 That isn't much of a reason. Its like saying we should stick to Assembly because its too much hard work to port the legacy code to C.

**Comment by towerofnix** [comment]
@Blaze349 I'm going with the assumption that legacy code *really does exist* for Python 2, though. *Nobody* uses code made in Assembly, but a lot of people used Python 2 code which makes Py2 kinda just stick around. (..this is just an assumption, please correct me, somebody who knows better..)

**Comment by benajaero** [comment]
@liam4 People used to use assembly for everything. People should move to Py 3. I don't see the benefits of going with Py 2.

**Comment by matthewr6** [comment]
@liam4 Your assumption is correct - I have experience with one.  (also @Blaze349 for this next part)

Python 3 is the way to go for new projects, sure, but plenty of old/existing projects are in Python 2, and it's not like there are huge differences.  Also, it can be a pain to port large projects... making sure everything still works the exact same way and stuff, for a few hundred thousand users

**Comment by benajaero** [comment]
https://docs.python.org/2/library/2to3.html

**Comment by matthewr6** [comment]
It's not like that's a magic solution if you're using multiple libraries and have a lot of things that could potentially break

**Comment by TheHockeyist** [comment]
Wait, so why are we arguing over if my (poorly written) code is Python 2 or 3?

**I JUST WANT MY CODE TO WORK, JEEZ!!!**

**Comment by benajaero** [comment]
@TheHockeyist If you want your code to work then follow liam4's instructions. Read the documentation and work through the examples.

**Comment by TheHockeyist** [comment]
Is there a cheat sheet or something?

**Comment by benajaero** [comment]
Yes. Its called documentation

**Comment by towerofnix** [comment]
I don't know of any good, quick-and-easy cheat sheets. You'll probably get the hang of the basics of programming with Python (that is, making programs that run 😛 ) pretty quickly. If you find yourself forgetting a specific piece of code a lot - for example, how to loop a number of times - you could write yourself a cheat sheet!

**Comment by TheHockeyist** [comment]
Well, the while loop works now. Not sure how to print.

**Comment by TheHockeyist** [comment]
Okay, I think I fixed the thing.

**Comment by TheHockeyist** [comment]
# **IT WORKS!!!!**
